src/v2/view-builder/views/duo/EnrollDuoAuthenticatorView.js

```diff
@@ -0,0 +1,44 @@
+import { loc, createCallout } from 'okta';
+import Duo from 'duo';
+import BaseForm from '../../internals/BaseForm';
+import BaseAuthenticatorView from '../../components/BaseAuthenticatorView';
+import AuthenticatorEnrollFooter from '../../components/AuthenticatorEnrollFooter';
+
+const Body = BaseForm.extend({
+
+  noButtonBar: true,
+
+  title  () {
+    return loc('oie.duo.enroll.title', 'login');
+  },
+
+  postRender: function () {
+    const contextualData = this.options.appState.get('currentAuthenticator').contextualData;
+    // This is the place to check contextualData.integrationType once we support more types
+    // Currently we only support IFRAME
+    const duoFrame = this.add(`<iframe frameborder="0" title="'${this.title()}'"></iframe>`).last();
+    try {
+      Duo.init({
+        host: contextualData.host,
+        sig_request: contextualData.signedToken, // eslint-disable-line camelcase
+        iframe: duoFrame.el,
+        post_action: (signedData) => { // eslint-disable-line camelcase
+          this.model.set('credentials.signatureData', signedData);
+          this.saveForm(this.model);
+        },
+      });
+    } catch (e) {
+      duoFrame.remove();
+      this.add(createCallout({
+        type: 'error',
+        subtitle: loc('oie.duo.iFrameError', 'login'),
+      }), '.o-form-error-container');
+      console.error(e); // eslint-disable-line no-console
+    }
+  },
+});
+
+export default BaseAuthenticatorView.extend({
+  Body,
+  Footer: AuthenticatorEnrollFooter,
+});
```

Comment:
The only difference between this file and the challenge is the title message. Would love for us to find a better way to render views conditionally based on a series of top-level attributes! I encountered a similar case with #1638.

Just a thought - no action needed from you at this time!

Reply:
One idea

```
// BaseDuoAuthencitorView.js
export BaseBody = BaseForm.extend({
  initialize() {}
});
export default BaseAuthenticatorView.extend({
  Footer: ...
});

// EnrollView.js

export default BaseDuoAuthencitorView.extend({
  Body: BaseBody.extend({
    title() {}
  });
});
```

Reply:
This was brought up before on this PR https://github.com/okta/okta-signin-widget/pull/1650#discussion_r572439321
I'll make the change 👍